### DISCUSSION TITLE
What are your thoughts on the new Temporal API?

### DISCUSSION THREAD

**Discussion by danantal** [js-fns/date-fns#2997] [General]
Hi,

This is more of a general question regarding your vision on the new [Temporal API proposal](https://github.com/tc39/proposal-temporal) and whether you are considering to start using it internally (similar to `Intl.DateTimeFormat`) in order to reduce the footprint even further as (and if) it gets adopted?

Or alternatively, where do you see that fit in? Would void some of the functionality? Do you recon the community should move towards a unified standard?

I'm curious on your thoughts!

**Comment by BasixKOR** [comment]
I just wanted to revisit this Q since the Temporal API is getting closed to finalized and started shipping to browsers. I would love to hear ideas and plans to adopt Temporal API into date-fns.

**Reply by jamesarosen** [reply]
It's possible to rewrite [formatDistance](https://github.com/date-fns/date-fns/blob/main/src/formatDistance/index.ts) using `Instant.prototype.until` (to create a `Duration`) and `Duration.prototype.round` to do the duration-aware rounding.
